src/js/user-profile/containers/UserProfileApp.jsx

```diff
@@ -1,40 +1,84 @@
 import React from 'react';
 import { connect } from 'react-redux';
 
+import ProgressButton from '../../common/components/form-elements/ProgressButton';
+import Modal from '../../common/components/Modal';
+
+import { removeSavedForm } from '../actions/index';
 import UserDataSection from '../components/UserDataSection';
 import AuthApplicationSection from '../components/AuthApplicationSection';
-
+import FormList from '../components/FormList';
 import RequiredLoginView from '../../common/components/RequiredLoginView';
 
+import { isSIPEnabledForm } from '../helpers';
+
 class UserProfileApp extends React.Component {
-  render() {
-    const view = (
-      <div className="row user-profile-row">
-        <div className="usa-width-two-thirds medium-8 small-12 columns">
-          <h1>Your Account</h1>
-          <div>
-            <AuthApplicationSection
-              userProfile={this.props.profile}
-              verifyUrl={this.props.verifyUrl}/>
-            <UserDataSection/>
-          </div>
-        </div>
-      </div>
-    );
-
-    return (
-      <div>
-        <RequiredLoginView
-          authRequired={1}
-          serviceRequired="user-profile"
-          userProfile={this.props.profile}
-          loginUrl={this.props.loginUrl}
-          verifyUrl={this.props.verifyUrl}>
-          {view}
-        </RequiredLoginView>
-      </div>
-    );
+  constructor(props) {
+    super(props);
+    this.state = { modalOpen: false };
   }
+
+ removeForm = () => {
+   this.toggleModal();
+   this.props.removeSavedForm(this.state.formId);
+ }
+
+ toggleModal = (formId) => {
+   this.setState({ formId, modalOpen: !this.state.modalOpen });
+ }
+
+ render() {
+   const savedForms = this.props.profile.savedForms;
+   const verifiedSavedForms = savedForms.filter(isSIPEnabledForm);
+   const hasVerifiedSavedForms = !!verifiedSavedForms.length;
+
+   const view = (
+     <div className="row user-profile-row">
+       <div className="usa-width-two-thirds medium-8 small-12 columns">
+         <h1>Your Account</h1>
+         <div>
+           {hasVerifiedSavedForms && <FormList
```

Comment:
I think I would move as much as possible of the logic you added in this component this into your `FormsList` component (or make another component that wraps that one). This is the main component for the user profile, so it would be nice to pare it down to calling a single component with a few props for the change we're making. It keeps this component from getting to cluttered with logic from different features in the profile.

Reply:
I see, thanks. Change made.